### PR TITLE
Auth/LDAP: More reliable load and syncup for LDAP attributes

### PR DESCRIPTION
**Pull request by indrajitr**
* Load user email when available
* Update descr and email if available when updating existing local user
* Add all the necessary search attributes while making LDAP query

NB: This is a followup of opnsense#3322

### CODE REVIEW

File: src/www/system_usermanager.php [hunk 1]
```diff
@@ -635,7 +635,7 @@ function get_user_privdesc(& $user)
                   <tr>
                     <td><a id="help_for_email" href="#" class="showhelp"><i class="fa fa-info-circle"></i></a> <?=gettext("E-Mail");?></td>
                     <td>
-                      <input name="email" type="text" value="<?= $pconfig['email'] ?>" />
+                      <input name="email" type="text" value="<?= $pconfig['email'] ?>" <?= $pconfig['scope'] == "system" || !empty($pconfig['user_dn']) ? "readonly=\"readonly\"" : "";?> />
```
Comment (AdSchellevis) [inline src/www/system_usermanager.php]: I don't think we should lock the email, if the other end doesn't return it, we likely get support questions about this.

File: src/www/system_usermanager_import_ldap.php [hunk 2]
```diff
@@ -30,14 +30,21 @@
...
-function add_local_user($username, $userdn, $userfullname)
+function add_local_user($username, $userdn, $userfullname, $useremail)
 {
     global $config;
 
     foreach ($config['system']['user'] as &$user) {
         if ($user['name'] == $username && $user['name'] != 'root') {
           // link local user to remote server by updating user_dn
           $user['user_dn'] = $userdn;
+          // also update descr and email if available
```
Comment (AdSchellevis) [inline src/www/system_usermanager_import_ldap.php]: please remove this block `local_user_set()` doesn't use these parameters.

File: src/www/system_usermanager.php (hunk 1 above)
Comment (indrajitr) [reply src/www/system_usermanager.php]: Agreed! We should probably have the same treatment for `descr` (fullname) as well, shouldn't we?

Comment (AdSchellevis) [reply src/www/system_usermanager.php]: that's debatable, it should be kept locked for system, but for ldap we might strip it off indeed

Comment (indrajitr) [reply src/www/system_usermanager.php]: Sure, so something like this works for `descr` (just removing check for empty `user_dn`)?

```
<input name="descr" type="text" value="<?=$pconfig['descr'];?>" <?= $pconfig['scope'] == "system" ? "readonly=\"readonly\"" : "";?> />
```

**Comment by indrajitr** [comment]
@AdSchellevis, all set. Thank you!

**Comment by AdSchellevis** [comment]
@indrajitr looks good, thanks!

**Pull request merged**